### ISSUE TITLE
[arch-update] Command '['checkupdates']' returned non-zero exit status 2.

### ISSUE THREAD

**Issue by MinecraftPlaye** [vivien/i3blocks-contrib#294]
### Expected behavior
Shows how many updates are available.

### Actual behavior
Does not show up on my i3blocks bar.

### i3blocks config relevant to blocklet(s)
```
# i3blocks configuration file
#
# The i3blocks man page describes the usage of the binary,
# and its website describes the configuration:
#
#     https://vivien.github.io/i3blocks


# Global properties
separator=true
separator_block_width=15

[battery2]
command=~/.i3blocks/battery2
markup=pango
interval=30

[documentation]
full_text=Docs
website=https://vivien.github.io/i3blocks
command=xdg-open "$website"
color=#f12711

[time]
command=date '+%Y-%m-%d %H:%M:%S'
interval=1

[pacman-updates]
command=~/.i3blocks/arch-update
markup=pango
interval= 3600
QUIET=true
WATCH=^linux.* ^pacman.*
BASE_COLOR=#5fff5f
UPDATE_COLOR=#FFFF85
AUR=true
LABEL= 
```

### Output of blocklet(s) when run from command line
```
Traceback (most recent call last):
  File "./arch-update", line 116, in <module>
    updates = get_updates()
  File "./arch-update", line 71, in get_updates
    raise exc
  File "./arch-update", line 67, in get_updates
    output = check_output(['checkupdates']).decode('utf-8')
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['checkupdates']' returned non-zero exit status 2.
```

### Output of any relevant other commands that might help diagnostics
`checkupdates | wc -l` returns 0.

### arch-update
```
#!/usr/bin/env python3
#
# Copyright (C) 2017 Marcel Patzwahl
# Licensed under the terms of the GNU GPL v3 only.
#
# i3blocks blocklet script to see the available updates of pacman and the AUR
import subprocess
from subprocess import check_output
import argparse
import os
import re


def create_argparse():
    def _default(name, default='', arg_type=str):
        val = default
        if name in os.environ:
            val = os.environ[name]
        return arg_type(val)

    strbool = lambda s: s.lower() in ['t', 'true', '1']
    strlist = lambda s: s.split()

    parser = argparse.ArgumentParser(description='Check for pacman updates')
    parser.add_argument(
        '-b',
        '--base_color',
        default = _default('BASE_COLOR', 'green'),
        help='base color of the output(default=green)'
    )
    parser.add_argument(
        '-u',
        '--updates_available_color',
        default = _default('UPDATE_COLOR', 'yellow'),
        help='color of the output, when updates are available(default=yellow)'
    )
    parser.add_argument(
        '-a',
        '--aur',
        action = 'store_const',
        const = True,
        default = _default('AUR', 'False', strbool),
        help='Include AUR packages. Attn: yay must be installed'
    )
    parser.add_argument(
        '-q',
        '--quiet',
        action = 'store_const',
        const = True,
        default = _default('QUIET', 'False', strbool),
        help = 'Do not produce output when system is up to date'
    )
    parser.add_argument(
        '-w',
        '--watch',
        nargs='*',
        default = _default('WATCH', arg_type=strlist),
        help='Explicitly watch for specified packages. '
        'Listed elements are treated as regular expressions for matching.'
    )
    return parser.parse_args()


def get_updates():
    output = ''
    try:
      output = check_output(['checkupdates']).decode('utf-8')
    except subprocess.CalledProcessError as exc:
      # usually this means no updates are available
      if not (exc.returncode == 1 and not exc.output):
        raise exc
    
    if not output:
        return []

    updates = [line.split(' ')[0]
               for line in output.split('\n')
               if line]

    return updates


def get_aur_updates():
    output = ''
    try:
        output = check_output(['yay', '-Qua']).decode('utf-8')
    except subprocess.CalledProcessError as exc:
        # yay exits with 1 and no output if no updates are available.
        # we ignore this case and go on
        if not (exc.returncode == 1 and not exc.output):
            raise exc
    if not output:
        return []

    aur_updates = [line.split(' ')[0] 
                  for line in output.split('\n')]


    return aur_updates


def matching_updates(updates, watch_list):
    matches = set()
    for u in updates:
        for w in watch_list:
            if re.match(w, u):
                matches.add(u)

    return matches


label = os.environ.get("LABEL","")
message = "{0}<span color='{1}'>{2}</span>"
args = create_argparse()

updates = get_updates()
if args.aur:
    updates += get_aur_updates()

update_count = len(updates)
if update_count > 0:
    info = str(update_count) + ' updates available'
    short_info = str(update_count) + ' updates'
    matches = matching_updates(updates, args.watch)
    if matches:
        info += ' [{0}]'.format(', '.join(matches))
        short_info += '*'
    print(message.format(label, args.updates_available_color, info))
    print(message.format(label, args.updates_available_color, short_info))
elif not args.quiet:
    print(message.format(label, args.base_color, 'system up to date'))
```

**Comment by MarcoMiano** [comment]
I have the same result as OP.
changing get_updates() as follow resolve the problem:
```
[...]
def get_updates():
    output = ''
    try:
      output = check_output(['checkupdates']).decode('utf-8')
    except subprocess.CalledProcessError as exc:
      # checkupdates exit with 2 and no output if no updates are available
      # we ignore this case and go on
      if not (exc.returncode == 2 and not exc.output):
        raise exc


    
    if not output:
        return []

    updates = [line.split(' ')[0]
               for line in output.split('\n')
               if line]

    return updates
[...]
```

checkpdates exit with 2 in case pacman (just check the end checkupdates script to find it out).

**Comment by flying-scorpio** [comment]
#319 addresses this. Will this merge ?

**Comment by jolange** [comment]
Thanks, it's fixed in f2715bc549453efaaf3a4998a9b6b418a3f044d4 now.
